RecoTracker/TkSeedGenerator/plugins/DeepCoreSeedGenerator.cc

```diff
@@ -0,0 +1,609 @@
+#include "Geometry/CommonDetUnit/interface/GlobalTrackingGeometry.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/Common/interface/DetSet.h"
+#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "DataFormats/JetReco/interface/Jet.h"
+#include "DataFormats/SiPixelDigi/interface/PixelDigi.h"
+#include "DataFormats/GeometryVector/interface/VectorUtil.h"
+#include "DataFormats/Math/interface/Point3D.h"
+#include "DataFormats/Math/interface/Vector3D.h"
+#include "DataFormats/Candidate/interface/Candidate.h"
+
+#include "RecoLocalTracker/ClusterParameterEstimator/interface/PixelClusterParameterEstimator.h"
+#include "RecoLocalTracker/Records/interface/TkPixelCPERecord.h"
+
+#include "TrackingTools/GeomPropagators/interface/StraightLinePlaneCrossing.h"
+#include "TrackingTools/GeomPropagators/interface/Propagator.h"
+#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
+
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include "SimDataFormats/Vertex/interface/SimVertex.h"
+
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+//
+// class declaration
+//
+struct DeepCoreCache {
+  mutable tensorflow::GraphDef* graph_def;
+};
+
+class DeepCoreSeedGenerator : public edm::stream::EDProducer<edm::GlobalCache<DeepCoreCache>> {
+public:
+  explicit DeepCoreSeedGenerator(const edm::ParameterSet&, const DeepCoreCache*);
+  ~DeepCoreSeedGenerator() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  // A pointer to a cluster and a list of tracks on it
+
+  // static methods for handling the global cache
+  static std::unique_ptr<DeepCoreCache> initializeGlobalCache(const edm::ParameterSet&);
+  static void globalEndJob(DeepCoreCache*);
+
+  struct TrackAndState {
+    TrackAndState(const reco::Track* aTrack, TrajectoryStateOnSurface aState) : track(aTrack), state(aState) {}
+    const reco::Track* track;
+    TrajectoryStateOnSurface state;
+  };
+
+  template <typename Cluster>
+  struct ClusterWithTracks {
+    ClusterWithTracks(const Cluster& c) : cluster(&c) {}
+    const Cluster* cluster;
+    std::vector<TrackAndState> tracks;
+  };
+
+  typedef ClusterWithTracks<SiPixelCluster> SiPixelClusterWithTracks;
+
+  // typedef std::vector<SiPixelClusterWithTracks> SiPixelClustersWithTracks;
```

Comment:
remove commented out code

Reply:
Am I wrong or at this point is possible to remove also the 
`typedef ClusterWithTracks<SiPixelCluster> SiPixelClusterWithTracks;`
and also the 
`  struct ClusterWithTracks {
`and the
`  struct TrackAndState {`
??

Reply:
yes, if you're not using them, they should be removed.